### DISCUSSION TITLE
Using commands in the scheduler

### DISCUSSION THREAD

**Discussion by hyperloop512** [KenT2/pipresents-gtk#6] [Q&A]
Hi,
Not really sure where to put such as discussion - cannot find anything on reddit, so:
Love Pipresents. Heaps of great documentation. I'm trying to figure out how to make a show work for my use case:

What I'd like to do is have a show that runs Monday-Friday and:
1. Turns a monitor on at 08:30. I tried putting "cec on" in the start show schedule, but it gives a validate error.
2. The show has a liveshow that I want to start every hour. I tried using the schedule tab for that show: (e.g. 09:00 openexclusive, 10:00 openexclusive, etc), but I can't use the openexclusive command - it errors. I also can't use the closeall command - so if the show contains liveshow1 which opens liveshow2 at the end, how do I get the scheduler to ensure that all shows are closed at the end of each hour?
3. How do I get it to turn the monitor off at night?

Many thanks for suggestions!

**Comment by hyperloop512** [comment]
Ah - so there are subshows...now trying to get them to start and stop on time is the next challenge.

**Comment by KenT2** [comment]
Discussions usually end up as Github Issues. I think Q&A is a relatively new facility

1. I will look at CEC ON in Start Show. Have you  installed cec-utils - sudo apt install cec-utils. You could put CEC ON in a normal show instead and schedule that show
2. Normal show scheduler only accepts open and close commands
3. CEC OFF command in start show or CEC OFF in a normal show that is set to run at night. Might need a dummy track in the show

**Reply by hyperloop512** [reply]
Thank you for the prompt reply. This is above and beyond the call of duty!

I am gradually getting PiPresents on an Rpi4 working, as I read the documentation more carefully. The documentation is impressive.

For turning the screen off and on automatically:
1 I tried using the other control sequence (xrandr in a normal MonitorOnOff show). It did turn the display output off, which is good, but when it turned it back on using xrandr the screen resolution was set to a very low value instead of the 4K it had been before. So I daren't use that.
2 I then tried the CEC commands in the MonitorOnOff show (it did complain about a missing dummy track). They didn't seem to have any effect sadly. Maybe it's because I'm using a 4K monitor and not a TV. I also tried running the cec scan tools from the command line, but it didn't seem to find the display connected to the RPi4. So I stopped there. 

For now I'll leave the screen black by just having the show end at the end of the day.

What I really like about PiPresents is that I can have a main show with two dubshows. The first presents a subshow of a series of webpages that are drawn from online sources (job adverts). The second subshow is a random sequence of images in a liveshow. This complete presentation runs on demand at specific times of day. (I was initially doing the webpages using a series of web browser commands, but it works almost as well with separate webkit tracks - the screen only blacks out for a second or two, and it is much easier to see what is going on in the web editor). 
I haven't quite worked out how to make the show run just Monday-Friday yet...it would be useful to have an ability in the web editor to simulate running through a show for a day or a week, and output the list of actions that will happen into a text file...in the meantime what is the best way to create a debug file for a show that is run fullscreen from an autostart. I only want to output the scheduled actions it has taken to the debug file, but I don't have a terminal window open, so am not sure what happens in this case...

Thanks again for the program - I'll post updates later in the week...

**Comment by KenT2** [comment]
The monitor commands seem to have been broken since RPi OS went to KMS/Wayland

To get a file of debug comands from autostart have a look in the file /pipresents/autostart.sh
